src/org/traccar/BaseEventHandler.java

```diff
@@ -12,24 +14,32 @@ public boolean isLastPosition() {
         return isLastPosition;
     }
 
+    private Device device;
```

Comment:
`device` is always retrieved in `handlePosition` for every position just to determine if it is last
and retrieved again in 2 of 4 event handlers in `analyzePosition` why not to decrease `getDeviceById` calls